### PR TITLE
Fluent-bit chart add splunk support

### PR DESCRIPTION
**Pull request by bryse-dev**
**What this PR does / why we need it**: This PR provides configuration for Fluent-Bit to send logs to Splunk.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by bryse-dev** [comment]
Signed the CLA

File: stable/fluent-bit/templates/config.yaml [hunk 1]
```diff
@@ -5,9 +5,6 @@ metadata:
   name: {{ template "fluent-bit.fullname" . }}-config
   labels:
     app: {{ template "fluent-bit.fullname" . }}
-    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
```
Comment (davidkarlsen) [inline stable/fluent-bit/templates/config.yaml]: why was this removed?

File: stable/fluent-bit/values.yaml [hunk 2]
```diff
@@ -53,13 +68,6 @@ backend:
     ## Can be either 'msgpack' or 'json'
     format: msgpack
 
-parsers:
```
Comment (davidkarlsen) [inline stable/fluent-bit/values.yaml]: why are these removed?

**Comment by mattfarina** [comment]
@kfox1111 @edsiper Can you please review this PR?

/ok-to-test

**Comment by kfox1111** [comment]
lgtm

**Comment by igrayson** [comment]
This PR broke the es backend config. See https://github.com/helm/charts/pull/7413

**Comment by robermorales** [comment]
This change moves the sub-conditions formerly inside the "es" sections, to be part of the "splunk" section. Breaking the connection with es.

File: stable/fluent-bit/templates/config.yaml [hunk 3]
```diff
@@ -70,6 +70,18 @@ data:
         Logstash_Format On
         Retry_Limit False
         Type  {{ .Values.backend.es.type }}
+{{ else if eq .Values.backend.type "splunk" }}
```
Comment (mwl) [inline stable/fluent-bit/templates/config.yaml:73]: By the looks of things, this if is terminated efter the `.Values.backend.es.*` ifs?

Comment (kfox1111) [reply stable/fluent-bit/templates/config.yaml:73]: Yeah, this breaks the .es.* options below. It should be moved down right above .Values.backend.type "http"

Comment (kfox1111) [reply stable/fluent-bit/templates/config.yaml:73]: nevermind. this review is old. it did in fact break things and we fixed it later.

Comment (mwl) [reply stable/fluent-bit/templates/config.yaml:73]: @kfox1111 Where has it been fixed? From what I can see it's still an issue on `master` and helm/charts#8166 isn't moving

Comment (kfox1111) [reply stable/fluent-bit/templates/config.yaml:73]: sigh. your right. we need to get some reviewers on that review asap.

Comment (kfox1111) [reply stable/fluent-bit/templates/config.yaml:73]: and this one was the one I was thinking merged: https://github.com/helm/charts/pull/7878

Comment (kfox1111) [reply stable/fluent-bit/templates/config.yaml:73]: and https://github.com/helm/charts/pull/7413 :/

Comment (kfox1111) [reply stable/fluent-bit/templates/config.yaml:73]: @unguiculus we have 3 different implementations of a bugfix on trunk due to lack of reviews and new pr's piling up for the same bug. how should we proceed?
